script/core/hint.lua

```diff
@@ -287,6 +287,8 @@ local function semicolonHint(uri, results, start, finish)
     ---@async
     guide.eachSourceTypes(state.ast, blockTypes, function (src)
         await.delay()
+        if #src < 1 then return end
```

Comment:
I think writing `== 0` is more common when checking empty array 😃 

Full searching codebase with
- regex `#\w+ == 0`
=> 260 results
- regex `#\w+ < 1`
=> 1 result only

Reply:
You're probably right, but using `<` results in fewer characters (which is why I did it 🤷‍♂️).

Reply:
> using `<` results in fewer characters

No offense, but **readability and consistency** are much more important than **fewer characters**. If **fewer characters** is a priority when writing code, why not just write `#src<1` without spaces? Even two characters fewer 😂 .

**Readability and consistency** are especially important in an open source project where there can be many contributors. If everyone just write with their own styles, then the project will become unmaintainable in no time. ☹️ 

---

Of course you can insist to write `< 1` as you are in charge in this PR, and maintainers may still thank you for the work and get it merged. But I just cannot agree with the reasoning behind...

Reply:
It's just inconsisten with the whole codebase
and it is just like 4 characters to change for you

Reply:
> I do not agree that my patch is unreadable lol

I didn't say that the current patch is unreadable, it's **inconsistent**. While I agree that these kind of things are trivial, many trivial things put together can have a huge impact. This is especially true in a collaborative environment like this one.

A random article about **consistency** searched in google:
https://dev.to/akashpattnaik/why-consistency-matters-the-most-in-programming-43ie

I believe every developer aims for excellence, and that often starts with attention to detail in seemingly minor aspects. 😃

Reply:
I think fixing this bug should involve adding a null check for `last` at line 317. The conditional logic here is quite complex in conjunction with the null checks below, which adds unnecessary cognitive load.

Reply:
> which adds unnecessary cognitive load.

Personally I think adding null check for `last`, on the other hand, makes this logic more complex. 😕 The current patch already early returns when array is empty, so any in-range index of this array will not be empty. This is the same that when we do `for i = 1, #src - 1 do`, as the index are in range, we need not to check `src[i] ~= nil` inside the loop.

If we still add nil check for `last`, it will add extra confusion that: "after checking array non empty", the in-range index array element can still be empty...

I originally proposed to check array non empty at line317 (https://github.com/LuaLS/lua-language-server/issues/2788#issuecomment-2280712436), but after seeing this patch I agree with @tmillr that we can return even earlier. 👍 

---

Of course you have the final call 😄